stubs/urllib3/urllib3/util/retry.pyi

```diff
@@ -12,16 +12,25 @@ ResponseError = exceptions.ResponseError
 log: Any
 
 class Retry:
-    DEFAULT_METHOD_WHITELIST: Any
-    BACKOFF_MAX: Any
-    total: Any
-    connect: Any
-    read: Any
-    redirect: Any
-    status_forcelist: Any
-    method_whitelist: Any
-    backoff_factor: Any
-    raise_on_redirect: Any
+    DEFAULT_ALLOWED_METHODS: ClassVar[frozenset[str]]
+    RETRY_AFTER_STATUS_CODES: ClassVar[frozenset[int]]
+    DEFAULT_REMOVE_HEADERS_ON_REDIRECT: ClassVar[frozenset[str]]
+    DEFAULT_BACKOFF_MAX: ClassVar[int]
+
+    total: bool | int | None
+    connect: int | None
+    read: int | None
+    redirect: bool | int | None
+    status: int | None
+    other: int | None
+    allowed_methods: Collection[str] | None
+    status_forcelist: Collection[int] | None
+    backoff_factor: float
+    raise_on_redirect: bool
+    raise_on_status: bool
+    history: tuple[Any] | None
```

Comment:
The attribute can never be `None` at runtime; if the `None` default in the `__init__` method is used, then the attribute is set to an empty tuple.

You'll need to add back the `RequestHistory` `NamedTuple` you had in one of your previous commits (it's a `collections.namedtuple` in the 1.26.x branch, but it can be a `typing.NamedTuple` in the stub, like it is in the `main` branch; type-checkers don't care about the different kinds of NamedTuples)

```suggestion
    history: tuple[()] | tuple[RequestHistory, ...]
```